golang/testsuite/testsuite_impl/exec_command_test/exec_command_test_.go

```diff
@@ -35,31 +36,28 @@ var execCommandThatShouldFail = []string{
 	"false",
 }
 
-type ExecCommandTest struct {}
+type ExecCommandTest struct{}
 
 func (e ExecCommandTest) Configure(builder *testsuite.TestConfigurationBuilder) {
 	builder.WithSetupTimeoutSeconds(30).WithRunTimeoutSeconds(30)
 }
 
 func (e ExecCommandTest) Setup(networkCtx *networks.NetworkContext) (networks.Network, error) {
+	return networkCtx, nil
+}
+
+func (e ExecCommandTest) Run(uncastedNetwork networks.Network) error {
+	// Necessary because Go doesn't have generics
+	castedNetworkContext := uncastedNetwork.(*networks.NetworkContext)
```

Comment:
Why are we moving the setup logic into Run?

Reply:
Mmm yeap, but maybe it's a mistake. We did on this way in `way_for_endpoint_availability` but maybe it shouldn't be this way on the remaining tests.

I'll return the addService to Setup method